content/6_Advanced/XOR_Basis.mdx

```diff
@@ -25,8 +24,214 @@ frequency: 1
 	>
 		used at USACO Camp
 	</Resource>
+	<Resource
+		source="hackmd"
+		url="https://hackmd.io/@minhnguyent546/BJKcH_2ti#XOR-BASIS"
+		title="Minh Thien Nguyen - XOR BASIS"
+		starred
+	>
+		detailed explanation about XOR basis note: translate the page to english
+	</Resource>
 </Resources>
 
+<FocusProblem problem="xorqueries" />
+
+You are given a tree consisting of $n$ vertices. There is an integer written on each vertex; the $i$-th vertex has integer $$a_i$$ written on it. You have to process $q$ queries. The $i$-th query consists of three integers $$x_i$$, $$y_i$$ and $$k_i$$. For this query, you have to answer if it is possible to choose a set of vertices $v1,v2,…,vm$ (possibly empty) such that:
+- every vertex $$v_j$$ is on the simple path between $$x_i$$ and $$y_i$$ (endpoints can be used as well);
+- $$a_{v_1} \oplus a_{v_2} \oplus \dots \oplus a_{v_m} = k_i$$, where $\oplus$ denotes the bitwise XOR operator.
+
+### Solution
+
+[Official Tutorial](https://codeforces.com/blog/entry/122951)
```

Comment:
yknow the point is to write an in-house explanation, not just link to some other thing

Reply:
added an in-house explanation to it. I have also commented the official tutorial, in case you want to include it for reference let me know